notebooks/04_model_select_and_optimize/aml_hyperparameter_tuning.ipynb

```diff
@@ -0,0 +1,619 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prerequisite\n",
+    "To run this example, you'll need to install [`azureml-sdk`](https://pypi.org/project/azureml-sdk/).\n",
+    "If you are using a [Data Science Virtual Machine (DSVM)](https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-configure-environment#dsvm) or [Azure Notebook](https://notebooks.azure.com/), `azureml-sdk` is already installed in it so you don't need to install the package.\n",
+    "\n",
+    "To install AML Python SDK*, run\n",
+    "```\n",
+    "pip install --upgrade azureml-sdk[notebooks]\n",
+    "```\n",
+    "\n",
+    "More info about setting up AML environment can be found from [this link](https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-configure-environment).\n",
+    "\n",
+    "> \\* AML has a Databricks sdk `azureml-sdk[databricks]` but it doesn't support hyperparameter tuning on Databricks for now."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### AML Workspace Configuration\n",
+    "AML workspace is the foundational block in the cloud that you use to experiment, train, and deploy machine learning models. We 1) setup a workspace from Azure portal and 2) create a config file manually. The instructions here are based on AML documents about [Quickstart with Azure portal](https://docs.microsoft.com/en-us/azure/machine-learning/service/quickstart-get-started) and [Quickstart with Python SDK](https://docs.microsoft.com/en-us/azure/machine-learning/service/quickstart-create-workspace-with-python) where you can find more details about the setup process with screen-shots.\n",
+    "  \n",
+    "<br>\n",
+    "  \n",
+    "#### Create a workspace\n",
+    "1. Sign in to the [Azure portal](https://portal.azure.com) by using the credentials for the Azure subscription you use.\n",
+    "2. Select **Create a resource** menu, search for **Machine Learning service workspace** select **Create** button.\n",
+    "3. In the **ML service workspace** pane, configure your workspace with entering the *workspace name* and *resource group* (or **create new** resource group if you don't have one already), and select **Create**. It can take a few moments to create the workspace.\n",
+    "  \n",
+    "<br>\n",
+    "  \n",
+    "#### Make a configuration file\n",
+    "To configure this notebook to communicate with your workspace easily, create a *.\\aml_config\\config.json* file with the following contents:\n",
```

Comment:
What about using `Workspace.write_config()` instead of doing this manually?

Reply:
Added that to the codes. Also updated the documentation markdown cells for that.